plugins/modules/volume.py

```diff
@@ -175,14 +186,99 @@ def poll_func() -> bool:
                 "failed to wait for volume status: timeout period expired"
             )
 
-    def _wait_for_volume_active(self) -> None:
+    def _wait_for_volume_active(self, volume: Volume) -> None:
```

Comment:
How about we make the `volume` param optional to avoid some code change when calling this method?

Reply:
The calls to `_wait_for_volume_active(self)` in the `_clone_volume` function require a volume parameter to work properly and the implementation of the function had to change slightly so that it would not work without the volume parameter. I tried changing it to optional and the tests all pass but I think that if we tried to make a call to `_wait_for_volume_active` without a volume parameter, even if it is optional, there would be an error.

Reply:
It's true that when cloning the volume we need the `volume` parameter, however, in other cases, i.e. `Ensure volume is active before continuing`, the param is not required. If we can make it optional, we don't have to change the legacy code or pass the volume param when it can be got from self. It can offer more flexibility and simplicity on how the function can be called. We can modify implementation of this function to tell that it can be used in both case. My intuitive thinking is like:

```
def _wait_for_volume_active(self, volume: Volume = None) -> None:
        # if volume is not passed, get its value from self (maybe also add the scenarios when it could be passed or not)
        if volume is None: 
             volume = self._volume
        self._wait_for_volume_status(
            volume, {"active"}, self._timeout_ctx.seconds_remaining
```

Let me know how you think about it!

Reply:
I tried this and this implementation caused errors. I'm not positive, but I think it was because self._volume is only populated with the correct volume in the function that calls _wait_for_volume_active rather than in _wait_for_volume_active itself. So, even though volume is set to self._volume when no argument is passed, self._volume is None at the point where it is assigned. This makes me think that the volume must be passed in the legacy calls because that's the only point at which we can access it through self._volume. What are your thoughts on this?

Reply:
I saw `self._volume` is assigned in line 246 or 248. So, how`self._volume` can be `None` when `_wait_for_volume_active` is call in line 252 and 256?

Reply:
Personally I'm in favor of the volume needing to be explicitly passed in as an argument since it requires that a volume must already exist before the function is called, which can improve readability and reduce ambiguity a bit. I think the issue we're running into right now is because of the fact that it's not clear whether or not the volume has been stored in `self._volume` at the time of calling. 

I originally wrote this module very early on in this project and at the time thought it'd be best to store the current volume in `self._volume`, but I think passing it around as a variable makes the code quite a bit cleaner. 👍

Reply:
> I saw `self._volume` is assigned in line 246 or 248. So, how `self._volume` can be `None` when `_wait_for_volume_active` is call in line 252 and 256?

This was happening after I made some code changes to make `volume` and optional parameter for `_wait_for_volume_active`. The code currently in the PR does not have this issue. Sorry for the confusion!